AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -217,207 +214,154 @@ - (void)readWithPartition:(NSString *)partition
+                       MSLogError([MSDataStore logTag], @"Not able to create document:%@", [cosmosDbError description]);
+                       completionHandler([[MSDocumentWrapper alloc] initWithError:cosmosDbError documentId:documentId]);
+                       return;
+                     }
+
+                     // Deserialize.
+                     NSError *deserializeError;
+                     NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:&deserializeError];
+                     if (deserializeError) {
+                       MSLogError([MSDataStore logTag], @"Error deserializing data:%@", [deserializeError description]);
+                     }
+                     MSLogDebug([MSDataStore logTag], @"Document json:%@", json);
+
+                     // Create an instance of Document.
+                     Class aClass = [document class];
+                     id<MSSerializableDocument> deserializedDocument =
+                         [(id<MSSerializableDocument>)[aClass alloc] initFromDictionary:(NSDictionary *)json[kMSDocumentKey]];
+
+                     // Create a document.
+                     NSTimeInterval interval = [(NSString *)json[kMSDocumentTimestampKey] doubleValue];
+                     NSDate *date = [NSDate dateWithTimeIntervalSince1970:interval];
+                     NSString *eTag = json[kMSDocumentEtagKey];
+                     MSDocumentWrapper *docWrapper = [[MSDocumentWrapper alloc] initWithDeserializedValue:deserializedDocument
+                                                                                                partition:partition
+                                                                                               documentId:documentId
+                                                                                                     eTag:eTag
+                                                                                          lastUpdatedDate:date];
+                     MSLogDebug([MSDataStore logTag], @"Document created with ID:%@", documentId);
+                     completionHandler(docWrapper);
+                     return;
+                   }];
 }
 
 - (void)deleteDocumentWithPartition:(NSString *)partition
                          documentId:(NSString *)documentId
                        writeOptions:(MSWriteOptions *)__unused writeOptions
                   completionHandler:(MSDataSourceErrorCompletionHandler)completionHandler {
+  [self performOperationForPartition:partition
+                          documentId:documentId
+                          httpMethod:kMSHttpMethodDelete
+                                body:[NSData data]
+                   additionalHeaders:nil
+                   completionHandler:^(NSData *__unused data, NSError *_Nonnull cosmosDbError) {
+                     // Body returned from call (data) is empty.
```

Comment:
nit: empty line

Reply:
Actually this is one of the unfortunate side effects of our formatter. When I add an empty line, the formatter removes it. Thus, if we commit it with the empty line, this will just come up the next time someone changes the file. In this case, we usually just keep the comment with no empty line before it.